src/app/clusters/webrtc-transport-requestor-server/webrtc-transport-requestor-server.cpp

```diff
@@ -0,0 +1,431 @@
+        // webRTCSessionID and store it to the CurrentSessions.
+        outSession.id         = sessionId;
+        outSession.peerNodeID = peerNodeId;
+        outSession.SetFabricIndex(peerFabricIndex);
+    }
+    else
+    {
+        // Also check that the existing session belongs to the same PeerNodeID / Fabric
+        // If it doesn’t match, respond with CONSTRAINT_ERROR
+        if (peerNodeId != existingSession->peerNodeID || peerFabricIndex != existingSession->GetFabricIndex())
+        {
+            ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::ConstraintError);
+            return;
+        }
+    }
+
+    std::string sdpAnswer(sdpSpan.data(), sdpSpan.size());
+
+    // Delegate handles Answer command received.
+    ctx.mCommandHandler.AddStatus(ctx.mRequestPath,
+                                  Protocols::InteractionModel::ClusterStatusCode(mDelegate.HandleAnswer(sessionId, sdpAnswer)));
+
+    // Store the new WebRTCSessionTypeStruct in CurrentSessions.
+    UpsertSession(outSession);
+}
+
+void WebRTCTransportRequestorServer::HandleICECandidates(HandlerContext & ctx, const Commands::ICECandidates::DecodableType & req)
+{
+    // Extract command fields from the request
+    uint16_t sessionId = req.webRTCSessionID;
+
+    std::vector<std::string> candidates;
+    auto iter = req.ICECandidates.begin();
+    while (iter.Next())
+    {
+        // Get current candidate CharSpan and convert to std::string.
+        const CharSpan & candidateSpan = iter.GetValue();
+        candidates.emplace_back(candidateSpan.data(), candidateSpan.size());
+    }
+
+    // Check the validity of the list.
+    CHIP_ERROR err = iter.GetStatus();
+    if (err != CHIP_NO_ERROR)
+    {
+        ChipLogError(Zcl, "HandleICECandidates: ICECandidates list error: %s", chip::ErrorStr(err));
+        ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Protocols::InteractionModel::Status::ConstraintError);
+        return;
+    }
+
+    // Check if the session, NodeID are valid
+    if (!IsPeerNodeSessionValid(sessionId, ctx))
+    {
+        ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::NotFound);
+        return;
+    }
+    // Handle ICE candidates in Delegate
+    ctx.mCommandHandler.AddStatus(
+        ctx.mRequestPath, Protocols::InteractionModel::ClusterStatusCode(mDelegate.HandleICECandidates(sessionId, candidates)));
+}
+
+void WebRTCTransportRequestorServer::HandleEnd(HandlerContext & ctx, const Commands::End::DecodableType & req)
+{
+    // Extract command fields from the request
+    uint16_t sessionId = req.webRTCSessionID;
+    auto reason        = req.reason;
+
+    // Validate the reason field against the allowed enum values.
+    if (reason == WebRTCEndReasonEnum::kUnknownEnumValue)
+    {
+        ChipLogError(Zcl, "HandleEnd: Invalid reason value %u.", static_cast<uint8_t>(reason));
+        ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Protocols::InteractionModel::Status::InvalidCommand);
```

Comment:
```suggestion
        ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Protocols::InteractionModel::Status::ConstraintError);
```
because this command is well-formed, but the value is out of range.

Reply:
Got it, done